server/remote/msgraph/get_default_calendar_view.go

```diff
@@ -73,7 +73,7 @@ func (c *client) DoBatchViewCalendarRequests(allParams []*remote.ViewCalendarPar
 		batchRes := &calendarViewBatchResponse{}
 		err := c.batchRequest(req, batchRes)
 		if err != nil {
-			c.changeUserStatus(err)
+			c.tokenHelpers.ChangeUserStatus(err, c.Logger, c.mattermostUserID, c.Poster)
```

Comment:
I think this will panic because `c.tokenHelpers` is not defined in this case. We are not using the user's token for the batch request, so there is no token to be expired

Reply:
I think we should make UserClient and SuperUserClient separate interfaces/structs, to separate the exact methods they implement. At the moment the SuperUserClient should only implement this one batch request method, while the UserClient would implement the rest of the client's methods currently